lib/sinon/sandbox.js

```diff
@@ -161,9 +161,21 @@ function Sandbox() {
     function getFakeRestorer(object, property) {
         var descriptor = getPropertyDescriptor(object, property);
 
-        return function restorer() {
+        function restorer() {
             Object.defineProperty(object, property, descriptor);
-        };
+        }
+        restorer.object = object;
+        restorer.property = property;
+        return restorer;
+    }
+
+    function verifyNotReplaced(object, property) {
+        for (var i = 0; i < fakeRestorers.length; i++) {
```

Comment:
Since we're targeting ES5.1, it's fine to use the ES5 Array Extras... it is much harder to create off-by-one errors, when you don't need to maintain an index 😉 

```js
function verifyNotReplaced(object, property) {
    fakeRestorers.forEach(function(fakeRestorer) {
        if (fakeRestorer.object === object && fakeRestorer.property === property) {
            throw new TypeError("Attempted to replace " + property + " which is already replaced");
        }
    });
}
```

Reply:
Yeah, you're right. And even `forEach` would give us an index as the second argument, if needed.